assets/chat/js/menus/ChatWhisperUsers.js

```diff
@@ -46,31 +70,37 @@ export default class ChatWhisperUsers extends ChatMenu {
 
   redraw() {
     this.updateNotification(); // its always visible
-    if (this.visible) {
-      this.usersEl.empty();
-      if (this.chat.whispers.size === 0) {
-        this.usersEl.append(this.empty);
-      } else {
-        [...this.chat.whispers.entries()]
-          .sort((a, b) => {
-            if (a[1].unread === 0) return 1;
-            if (b[1].unread === 0) return -1;
-            return 0;
-          })
-          .forEach((e) => this.addConversation(e[0], e[1].unread));
-      }
+    this.unreadEl.empty();
+    this.readEl.empty();
+    if (this.chat.whispers.size > 0) {
+      [...this.chat.whispers.values()]
+        .sort((a, b) => new Date(b.time).getTime() - new Date(a.time).getTime())
+        .forEach((whisper) => this.addConversation(whisper));
     }
+    this.ui.toggleClass('search-in', this.searchterm.length > 0);
     super.redraw();
   }
 
-  addConversation(nick, unread) {
-    const user = this.chat.users.get(nick.toLowerCase()) || new ChatUser(nick);
-    this.usersEl.append(`
-            <li class="conversation unread-${unread}">
-                <a style="flex: 1;" data-username="${user.username}" class="user">${user.displayName}</a>
-                <span class="badge">${unread}</span>
-                <a data-username="${user.username}" title="Hide" class="remove"></a>
-            </li>
-        `);
+  addConversation(whisper) {
+    const time = moment.utc(whisper.time).local();
+    const found = whisper.found && this.searchterm.length > 0 ? ' found' : '';
+    const online = this.chat.users.has(whisper.nick.toLowerCase())
+      ? 'online'
+      : 'offline';
+    const unread =
+      whisper.unread > 0
+        ? `<span class="unread">${whisper.unread} new</span>`
+        : '';
+
+    (whisper.unread > 0 ? this.unreadEl : this.readEl).append(`
+    <div class="user-entry ${online}${found}" title="${online}" data-username="${whisper.nick.toLowerCase()}">
+      <span class="user">${whisper.nick}</span>
+      <div class="right">
+        ${unread}
+        <span class="time" title="${time.format(
+          'LLL',
+        )}">${time.fromNow()}</span>
```

Comment:
Would you mind using code tags to make this a little more readable? It's a dependency we added sometime after this PR.